host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -1790,6 +1833,70 @@ esp_err_t uac_host_device_open(const uac_host_device_config_t *config, uac_host_
     return ret;
 }
 
+esp_err_t uac_host_device_open_with_vid_pid(uint16_t vid, uint16_t pid, const uac_host_device_config_t *config,
+        uint32_t timeout, uac_host_device_handle_t *uac_dev_handle)
+{
+    UAC_RETURN_ON_INVALID_ARG(uac_dev_handle);
+    UAC_RETURN_ON_INVALID_ARG(config);
+    UAC_RETURN_ON_FALSE(config->iface_num, ESP_ERR_INVALID_ARG, "Invalid interface number");
+    UAC_RETURN_ON_FALSE(s_uac_driver, ESP_ERR_INVALID_STATE, "UAC Driver is not installed");
+    UAC_RETURN_ON_FALSE(vid, ESP_ERR_INVALID_ARG, "Invalid vendor ID");
+    UAC_RETURN_ON_FALSE(pid, ESP_ERR_INVALID_ARG, "Invalid product ID");
+
+    uint32_t timeout_cnt = 0;
+    do {
+        // current time in ticks
+        uint32_t timeout_start = xTaskGetTickCount();
+        // check if the device is already opened
+        *uac_dev_handle = get_interface_by_vid_pid(vid, pid, config->iface_num);
+        if (*uac_dev_handle) {
+            ESP_LOGD(TAG, "Device VID 0x%04X, PID 0x%04X, Interface %d already opened", vid, pid, config->iface_num);
+            return ESP_OK;
+        }
+
+        // check all connected devices
+        int actual_count = 0;
+        uint8_t dev_addr_list[6] = {0};
```

Comment:
Const variable `6` better to make configurable via manuconfig. 

Something like: 

```suggestion
        uint8_t dev_addr_list[CONFIG_UAC_DEV_ADDR_LIST_MAX] = {0};
```